hoomd/md/PCNDAngleForceCompute.cc

```diff
@@ -0,0 +1,256 @@
+    const BoxDim& box = m_pdata->getGlobalBox();
+   
+    //uint64_t PCNDtimestep = 0;
+    //uint16_t seed = m_sysdef->getSeed();
+
+    // for each of the angles
+    const unsigned int size = (unsigned int)m_pcnd_angle_data->getN();
+    for (unsigned int i = 0; i < size; i++)
+        {
+        // lookup the tag of each of the particles participating in the angle
+	const AngleData::members_t& angle = m_pcnd_angle_data->getMembersByIndex(i);
+	assert(angle.tag[0] <= m_pdata->getMaximumTag());
+	assert(angle.tag[1] <= m_pdata->getMaximumTag());
+	assert(angle.tag[2] <= m_pdata->getMaximumTag());
+        
+        // transform a, b, and c into indicies into the particle data arrays
+	// MEM TRANSFER: 6 ints
+	unsigned int idx_a = h_rtag.data[angle.tag[0]];	
+	unsigned int idx_b = h_rtag.data[angle.tag[1]];	
+	unsigned int idx_c = h_rtag.data[angle.tag[2]];
+
+        // throw an error if this angle is incomplete
+	if (idx_a == NOT_LOCAL || idx_b == NOT_LOCAL || idx_c == NOT_LOCAL)
+	    {
+	    this->m_exec_conf->msg->error()
+	        << "angle.pcnd: angle " << angle.tag[0] << " " << angle.tag[1] << " "
+		<< angle.tag[2] << " incomplete." << endl
+		<< endl;
+            throw std::runtime_error("Error in PCND calculation");
+	    }
+
+	assert(idx_a < m_pdata->getN() + m_pdata->getNGhosts());
+	assert(idx_b < m_pdata->getN() + m_pdata->getNGhosts());
+	assert(idx_c < m_pdata->getN() + m_pdata->getNGhosts());
+        
+	//unsigned int j = m_group->getMemberIndex(i);
+	//unsigned int ptag = h_tag.data[j];
+	uint16_t seed = 1;
+	unsigned int angle_type = m_pcnd_angle_data->getTypeByIndex(i);
+
+	// Initialize the Random Number Generator and generate the 6 random numbers
+	RandomGenerator rng(hoomd::Seed(RNGIdentifier::PCNDAngleForceCompute, timestep, seed),
+			    hoomd::Counter(0));
+	UniformDistribution<Scalar> uniform(Scalar(0), Scalar(1));
+
+	Scalar a_x = uniform(rng);
+	Scalar b_x = uniform(rng);
+	Scalar a_y = uniform(rng);
+	Scalar b_y = uniform(rng);
+	Scalar a_z = uniform(rng);
+	Scalar b_z = uniform(rng);
+        	
+	if (idx_b < m_pdata->getN() && PCNDtimestep == 0)
+	   {
+	   h_force.data[idx_b].x = m_Xi[angle_type] * sqrt(-2 * log(a_x)) * cos(2 * 3.1415926535897 * b_x);
+	   h_force.data[idx_b].y = m_Xi[angle_type] * sqrt(-2 * log(a_y)) * cos(2 * 3.1415926535897 * b_y);
+	   h_force.data[idx_b].z = m_Xi[angle_type] * sqrt(-2 * log(a_z)) * cos(2 * 3.1415926535897 * b_z);
+	
+	   h_force.data[idx_b].w = sqrt(h_force.data[idx_b].x * h_force.data[idx_b].x + h_force.data[idx_b].y * h_force.data[idx_b].y + h_force.data[idx_b].z * h_force.data[idx_b].z);
+	   }
+
+	else if (idx_b < m_pdata->getN() && PCNDtimestep != 0)
+           {
+	   Scalar magx = h_force.data[idx_b].x;
+	   Scalar magy = h_force.data[idx_b].y;
+	   Scalar magz = h_force.data[idx_b].z;
+
+	   Scalar E = exp(-1 / m_Tau[angle_type]);
+	   Scalar hx = m_Xi[angle_type] * sqrt(-2 * (1 - E * E) * log(a_x)) * cos(2 * 3.1415926535897 * b_x);
```

Comment:
Yes, feel use `NormalDistribution` provided in `RandomNumbers.h`. It is well tested and working.